pallets/delegation/src/lib.rs

```diff
@@ -965,6 +974,56 @@ pub mod pallet {
 			removals = removals.saturating_add(1);
 			Ok((removals, consumed_weight))
 		}
+
+		#[cfg(any(feature = "try-runtime", test))]
+		pub fn do_try_state() -> Result<(), &'static str> {
+			fn get_merged_subtree<T: Config>(node: DelegationNode<T>) -> sp_std::vec::Vec<DelegationNode<T>> {
+				let mut nodes_to_explore = sp_std::vec::Vec::from([node]);
+				let mut children = sp_std::vec::Vec::new();
+				while let Some(current_node) = nodes_to_explore.pop() {
+					let child_nodes = current_node.children.iter().filter_map(DelegationNodes::<T>::get);
+					nodes_to_explore.extend(child_nodes.clone());
+					children.extend(child_nodes);
+				}
+				children
+			}
+
+			DelegationNodes::<T>::iter().try_for_each(
+				|(delegation_node_id, delegation_details)| -> Result<(), &'static str> {
+					let hierarchy_id = delegation_details.hierarchy_root_id;
+
+					// check if node is in part of a delegation hierarchy.
+					ensure!(
+						DelegationHierarchies::<T>::contains_key(hierarchy_id),
+						"Unknown hierarchy"
+					);
+
+					let parent_count = DelegationNodes::<T>::iter_values()
+						.filter(|delegation_node: &DelegationNode<T>| {
+							delegation_node.children.contains(&delegation_node_id)
+						})
+						.count();
+
+					match delegation_details.parent {
+						// If node is a leaf or intermediate, check if it occurs only once. Otherwise we have cycles.
```

Comment:
Would an `if let Some(...)` be better here?

Reply:
Or if we don't need the value in the `Some` variant, even just `if delegation_details.parent.is_some()` is enough.